fast.py

```diff
@@ -1,58 +1,142 @@
+"""Load English-language names for FAST and LCSH subject classifications
+from N-Triple files acquired from data dumps.
+
+This is how we know, e.g. that FAST classification 1750175 means
+"Short stories, American".
+"""
+from contextlib import contextmanager
 import gzip
 import csv
+from io import BytesIO
 import logging
 import os
 import re
 from nose.tools import set_trace
 import time
+import zipfile
 
 class FASTNames(dict):
 
     SUBDIR = "FAST"
 
     triple_re = re.compile('^<http://id.worldcat.org/fast/([0-9]+)> <http://schema.org[#/]name> "([^"]+)"')
 
-    def load_filehandle(self, fh):
-        for triple in fh:
-            triple = triple.strip()
-            g = self.triple_re.search(triple)
-            if not g:
-                continue
-            identifier, name = g.groups()
-            self[identifier] = name
-
     @classmethod
     def from_data_directory(cls, data_directory):
+        """Load names from a directory that either contains a bunch of
+        files in N-Triples format or a single consolidated CSV file.
```

Comment:
Are N-Triple files only handled here from this data source in all of the project?

Reply:
Yeah, that's the format used in the data dumps. We deal with RDF data in the OCLC Linked Data code, but it's in JSON-LD format. And even here we're not doing proper N-Triple processing, just scanning for certain regular expressions.